pkg/logql/log/pipeline.go

```diff
@@ -227,6 +230,9 @@ func (p *streamPipeline) Process(ts int64, line []byte, structuredMetadata ...la
 
 	for i, lb := range structuredMetadata {
 		structuredMetadata[i].Name = prometheus.NormalizeLabel(lb.Name)
+		if strings.ContainsRune(structuredMetadata[i].Value, utf8.RuneError) {
+			structuredMetadata[i].Value = string(bytes.Map(removeInvalidUtf, []byte(structuredMetadata[i].Value)))
```

Comment:
should we use `strings.Map` to avoid the additional copy when converting byte slice to string?